internal/nfdutil/nfdutil.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfdutil
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	nfdv1alpha1 "sigs.k8s.io/node-feature-discovery/api/nfd/v1alpha1"
+)
+
+// CheckNodeFeatureRule checks if a NodeFeatureRule exists that is owned by the NIM operator
+func CheckNodeFeatureRule(ctx context.Context, k8sClient client.Client) (bool, error) {
+	// Get operator namespace from the environment
+	namespace := os.Getenv("OPERATOR_NAMESPACE")
+	if namespace == "" {
+		return false, fmt.Errorf("OPERATOR_NAMESPACE is not set")
+	}
+
+	// Check if the NodeFeatureRule CRD exists
+	crdExists, err := checkCRDExists(ctx, k8sClient, "nodefeaturerules.nfd.k8s.io")
+	if err != nil {
+		return false, err
+	}
+
+	if !crdExists {
+		return false, nil
```

Comment:
return a new error saying that NFD CRD does not exist instead of nil?

Reply:
the intent of this function is to check if a specific NodeFeatureRule CR exists, so if the CRD itself is missing we can assume CR is not present as well. Hence error is not thrown in this case.